client/share_trader/js/components/share_trader.js

```diff
@@ -50,26 +47,61 @@ const ShareTrader = React.createClass({
         this.setState(state);
     },
 
-    setActiveAccount(account) {
-        this.setState({
-            activeAccount: account
+    fetchAssetList(account) {
+        AssetActions.flushAssetList();
+        const { activeAccount } = this.state;
+
+        /* fetch all assets */
+        let accountPublicKey = null;
+
+        if (account || activeAccount) {
+            accountPublicKey = account ? account.vk : activeAccount.vk;
+        }
+        accountPublicKey = account === null ? null : accountPublicKey;
```

Comment:
Puhhh, that's a though one.

Well, for starters we could try treating `undefined` and `null` as the same thing. Not sure if that's worth the time though.